src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -53,7 +59,26 @@ public class VsPackageInstaller : IVsPackageInstaller2
 
             _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
 
-            PumpingJTF = new PumpingJTF(NuGetUIThreadHelper.JoinableTaskFactory.Context);
+            PumpingJTF = null;
+        }
+
+        private void RunJTFWithCorrectContext(Project project, Func<Task> asyncTask)
+        {
+            if (PumpingJTF == null)
+            {
+                var VsHierarchy = VsHierarchyItem.FromDteProject(project).VsHierarchy;
```

Comment:
Why do you need to create `VsHierarchyItem` here? can't you just directly use `VsHierarchyUtility.ToVsHierarchy(project)`